util/site-dashboard/ot-nightly-results.js

```diff
@@ -2,67 +2,87 @@
+    "i2c": ["https://reports.opentitan.org/hw/ip/i2c/dv/latest/"],
+    "keymgr": ["https://reports.opentitan.org/hw/ip/keymgr/dv/latest/"],
+    "kmac": ["https://reports.opentitan.org/hw/ip/kmac_masked/dv/latest/", "https://reports.opentitan.org/hw/ip/kmac_unmasked/dv/latest/"],
+    "lc_ctrl": ["https://reports.opentitan.org/hw/ip/lc_ctrl/dv/latest/"],
+    "otbn": ["https://reports.opentitan.org/hw/ip/otbn/dv/uvm/latest/"],
+    "otp_ctrl": ["https://reports.opentitan.org/hw/ip/otp_ctrl/dv/latest/"],
+    "pattgen": ["https://reports.opentitan.org/hw/ip/pattgen/dv/latest/"],
+    "prim_alert": ["https://reports.opentitan.org/hw/ip/prim/dv/prim_alert/latest/"],
+    "prim_esc": ["https://reports.opentitan.org/hw/ip/prim/dv/prim_esc/latest/"],
+    "prim_lfsr": ["https://reports.opentitan.org/hw/ip/prim/dv/prim_lfsr/latest/"],
+    "prim_present": ["https://reports.opentitan.org/hw/ip/prim/dv/prim_present/latest/"],
+    "prim_prince": ["https://reports.opentitan.org/hw/ip/prim/dv/prim_prince/latest/"],
+    "pwm": ["https://reports.opentitan.org/hw/ip/pwm/dv/latest/"],
+    "pwrmgr": ["https://reports.opentitan.org/hw/ip/pwrmgr/dv/latest/"],
+    "rom_ctrl": ["https://reports.opentitan.org/hw/ip/rom_ctrl/dv/latest/"],
+    "rstmgr": ["https://reports.opentitan.org/hw/ip/rstmgr/dv/latest/"],
+    "rstmgr_cnsty_chk": ["https://reports.opentitan.org/hw/ip/rstmgr/dv/rstmgr_cnsty_chk/latest/"],
+    "rv_dm": ["https://reports.opentitan.org/hw/ip/rv_dm/dv/latest/"],
+    "rv_timer": ["https://reports.opentitan.org/hw/ip/rv_timer/dv/latest/"],
+    "spi_device": ["https://reports.opentitan.org/hw/ip/spi_device/dv/latest/"],
+    "spi_host": ["https://reports.opentitan.org/hw/ip/spi_host/dv/latest/"],
+    "sram_ctrl": ["https://reports.opentitan.org/hw/ip/sram_ctrl_main/dv/latest/", "https://reports.opentitan.org/hw/ip/sram_ctrl_ret/dv/latest/"],
+    "sysrst_ctrl": ["https://reports.opentitan.org/hw/ip/sysrst_ctrl/dv/latest/"],
+    "tl_agent": ["https://reports.opentitan.org/hw/dv/sv/tl_agent/dv/latest/"],
+    "uart": ["https://reports.opentitan.org/hw/ip/uart/dv/latest/"],
+    "usbdev": ["https://reports.opentitan.org/hw/ip/usbdev/dv/latest/"],
+    "xbar": ["https://reports.opentitan.org/hw/top_earlgrey/ip/xbar_main/dv/autogen/latest/", "https://reports.opentitan.org/hw/top_earlgrey/ip/xbar_peri/dv/autogen/latest/"],
+    "rv_core_ibex": ["https://ibex.reports.lowrisc.org/opentitan/latest/"],
+};
 
-async function fetch_block_report(block_report_url) {
+const chip_level_urls = {
+    "chip_earlgrey": ["https://reports.opentitan.org/hw/top_earlgrey/dv/latest/"],
+};
+
+/**
+ * @param {string} name - Optional name to override the block name pulled from the report.
+ * @param {string} report_url - Url of the report to be fetched.
+ * @returns {BlockLevelResults} Object containing report data after some processing.
+ */
+async function fetch_and_process_report(name = undefined, report_url) {
```

Comment:
This is very confusing. JavaScript has no notation of named parameters. So the `= undefined` will only be used when calling the function with 0 arguments but then report_url will also be undefined.

Reply:
Yeah good point the argument order should be swapped here.

Reply:
I don't think the `= undefined` is needed at all.